### PR TITLE
[Impeller] Compute correct-ish binding order for Vulkan.

### PR DESCRIPTION
**Pull request by jonahwilliams**
Workaround for https://github.com/flutter/flutter/issues/150324

Fixing binding offset of UBO if its not the first binding.
Tries to infer binding offset for sampler2D, but this may be incorrect if there is more than one sampler.

### CODE REVIEW

**Review by chinmaygarde — APPROVED** [review]

File: impeller/compiler/reflector.cc [hunk 1]
```diff
@@ -376,6 +376,8 @@ std::shared_ptr<RuntimeStageData::Shader> Reflector::GenerateRuntimeStageData()
 
     const auto& ubo = ubos[0];
 
+    size_t binding =
```
[the remark this answers is not in the document]
Comment (jonahwilliams) [reply impeller/compiler/reflector.cc:379]: its still broken though :)

**Comment by jonahwilliams** [comment]
agh. This is running on Vulkan but I need runtime stage vulkan.

File: impeller/fixtures/uniforms_and_sampler.frag [hunk 2]
```diff
@@ -0,0 +1,13 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// sampler is specifically ordered before color.
+uniform sampler2D u_texture;
```
[the remark this answers is not in the document]
Comment (jonahwilliams) [reply impeller/fixtures/uniforms_and_sampler.frag]: Done

File: impeller/runtime_stage/runtime_stage_unittests.cc [hunk 3]
```diff
@@ -225,6 +225,37 @@ TEST_P(RuntimeStageTest, CanReadUniforms) {
...
+TEST_P(RuntimeStageTest, CanReadUniformsUBOAndSampler) {
+  const std::shared_ptr<fml::Mapping> fixture =
+      flutter::testing::OpenFixtureAsMapping("uniforms_and_sampler.frag.iplr");
+  ASSERT_TRUE(fixture);
+  ASSERT_GT(fixture->GetSize(), 0u);
+  auto stages = RuntimeStage::DecodeRuntimeStages(fixture);
+  auto stage = stages[PlaygroundBackendToRuntimeStageBackend(GetBackend())];
+
+  ASSERT_TRUE(stage->IsValid());
+  switch (GetBackend()) {
```
[the remark this answers is not in the document]
Comment (jonahwilliams) [reply impeller/runtime_stage/runtime_stage_unittests.cc]: Done

**Pull request merged**